hashprobe/aeshash/aeshash_amd64.s

```diff
@@ -0,0 +1,150 @@
+//go:build !purego
+
+#include "textflag.h"
+
+// func Hash32(value uint32, seed uintptr) uintptr
+TEXT ·Hash32(SB), NOSPLIT, $0-24
+    MOVQ value+0(FP), AX
+    MOVQ seed+8(FP), BX
+
+    MOVOU runtime·aeskeysched+0(SB), X1
+    MOVOU runtime·aeskeysched+16(SB), X2
+    MOVOU runtime·aeskeysched+32(SB), X3
+
+    MOVQ BX, X0
+    PINSRD $2, AX, X0
+
+    AESENC X1, X0
+    AESENC X2, X0
+    AESENC X3, X0
+
+    MOVQ X0, ret+16(FP)
+    RET
+
+// func Hash64(value uint64, seed uintptr) uintptr
+TEXT ·Hash64(SB), NOSPLIT, $0-24
+    MOVL value+0(FP), AX
```

Comment:
Should it be `MOVQ` here and `MOVL` when loading the `value` in `Hash32` above?

Reply:
```
$ go vet ./hashprobe/...
# github.com/segmentio/parquet-go/hashprobe/aeshash
hashprobe/aeshash/aeshash_amd64.s:7:1: [amd64] Hash32: invalid MOVQ of value+0(FP); uint32 is 4-byte value
hashprobe/aeshash/aeshash_amd64.s:26:1: [amd64] Hash64: invalid MOVL of value+0(FP); uint64 is 8-byte value
```
👍